### PR TITLE
ui for foster care card

### PR DESCRIPTION
**Pull request by lauherkcambria**
#### What tests were added (if relevant), and do all tests (rspec, karma, and integration CALS tests) pass?
- foster care karma tests up to 60%

#### Which User story does this relate to?
- https://www.pivotaltracker.com/story/show/147228003
-https://www.pivotaltracker.com/story/show/147954691

#### Brief feature description
- feature development for foster care card with api connections

#### Does this PR block/effect other stories?
-no

#### Do any commands needs to be run in order to run the project with this PR?
-

#### Code coverage percent diff as a result of this PR?
-
left todo
#### JS errors diff as a result of this PR?
-
left todo

### CODE REVIEW

**Comment by ratneshraval** [comment]
Can you fill out story info and add tests?

File: app/javascript/forms/index.js [hunk 1]
```diff
@@ -11,6 +12,12 @@ export default class Forms extends React.Component {
     super(props)
     this.state = {
       focusComponentName: '',
+      visibleFields: {
+        visibleAgencyName: false,
+        visibleTypeOfLicense: false,
```
Comment (ratneshraval) [inline app/javascript/forms/index.js]: The visibility of the fields can be calculated from the option values we have in state. So this shouldn't be in state.

File: app/javascript/forms/addressCard.js [hunk 2]
```diff
@@ -79,8 +79,8 @@ export default class AddressCard extends React.Component {
             <DropDownField gridClassName='col-md-6' selectClassName='reusable-select'
               text={this.props.physicalMailingSimilar}
               optionList={yesNo.items}
-              label='Mailing address the same as Physical Address?'
-              onChange={(event) => this.props.setParentState('physical_mailing_similar', event.target.selectedOptions[0].value)} />
+              label={'Mailing address the same as Physical Address?'}
+              onChange={(event, number) => this.onChange(event.target.selectedOptions[0].text, ('physical_mailing_similar'))} />
```
Comment (ratneshraval) [inline app/javascript/forms/addressCard.js]: This is somehow overriding the correct code in development. Probably created during merge.

Comment (lauherkcambria) [inline app/javascript/forms/addressCard.js]: ah, is the onChange method supposed to be the this.props.setParentState method?

File: app/javascript/forms/index.js (hunk 1 above)
Comment (lauherkcambria) [inline app/javascript/forms/index.js]: i'm not quite sure i follow, can you show me?

File: app/javascript/forms/addressCard.js (hunk 2 above)
Comment (ratneshraval) [inline app/javascript/forms/addressCard.js]: Yes.

**Comment by lauherkcambria** [comment]
07/24/2017: left TODO: finish update to variable names to match contract saving, and adding enzyme test cases

**Comment by lauherkcambria** [comment]
hey @ratneshraval @anuroop15 if you have any improvements that can improve my tests cases please let me know! 🕺

**Comment by lauherkcambria** [comment]
`@lauherkcambria UI doesn't match the design.
Fields should be 1/3rd of 12 column Bootstrap layout. But it seems to be 2/3rd.

Also,
added card's doesn't align. PFB screenshot
screen shot 2017-08-02 at 11 58 23 am`

@anuroop15 this was by design to line up with the UI document.

**Comment by ratneshraval** [comment]
If you bring #139 in, it will probably help with clearing card issue.

**Comment by lauherkcambria** [comment]
`@lauherkcambria: It's not clearing the input and dropdown fields entered, when close/clear button was clicked`

@anuroop15 sorry anuroop, i just saw your comments, i don't know why they didn't show up and are outdated. thanks for the feedback, i'll address them once authentication is fixed. 👍

**Comment by lauherkcambria** [comment]
@anuroop15 fixed the alignment issue
@ratneshraval i got the javascript methods and save/connect api to work without refactoring in the addcard as js and remove card as js methods -- i'll look into that refactor and adding some more tests this afternoon. Also, the ao1 controller needs a serious refactor. srlsy 4rlz 👍

**Comment by ratneshraval** [comment]
I added removeCardJS method but addCard is causing some tests to fail even though it is functioning fine. I am approving this but can you start another PR to update test cases where are are also testing methods/data.

**Comment by lauherkcambria** [comment]
if changes break tests, shouldn't we fix those first?

**Pull request merged**